server/container.go

```diff
@@ -411,10 +411,50 @@ func (s *Server) RemoveContainer(ctx context.Context, req *pb.RemoveContainerReq
 	return &pb.RemoveContainerResponse{}, nil
 }
 
+// filterContainer returns whether passed container matches filtering criteria
+func filterContainer(c *pb.Container, filter *pb.ContainerFilter) bool {
+	if filter != nil {
+		if filter.State != nil {
+			if *c.State != *filter.State {
+				return false
+			}
+		}
+		// TODO(mrunalp): Add support for label filtering
+	}
+	return true
+}
+
 // ListContainers lists all containers by filters.
 func (s *Server) ListContainers(ctx context.Context, req *pb.ListContainersRequest) (*pb.ListContainersResponse, error) {
 	var ctrs []*pb.Container
-	for _, ctr := range s.state.containers.List() {
+	filter := req.Filter
+	ctrList := s.state.containers.List()
+
+	// Filter using container id and pod id first.
+	if filter != nil {
+		if filter.Id != nil {
+			c := s.state.containers.Get(*filter.Id)
+			if c != nil {
+				if filter.PodSandboxId != nil {
+					if c.Sandbox() == *filter.PodSandboxId {
+						ctrList = []*oci.Container{c}
+					} else {
+						ctrList = []*oci.Container{}
+					}
+
+				} else {
+					ctrList = []*oci.Container{c}
+				}
+			}
+		} else {
+			if filter.PodSandboxId != nil {
+				pod := s.state.sandboxes[*filter.PodSandboxId]
```

Comment:
is there a `Get` method in `sandboxes` as in `containers`? if not should you check `pod == nil` here?